app_dart/lib/src/request_handlers/push_build_status_to_github.dart

```diff
@@ -6,122 +6,140 @@ import 'dart:async';
-    final BuildStatusService buildStatusService = buildStatusServiceProvider(datastore);
-    final RepositorySlug slug = config.flutterSlug;
-    final GithubService githubService = await config.createGithubService(slug);
-
     if (authContext.clientContext.isDevelopmentEnvironment) {
       // Don't push GitHub status from the local dev server.
       return Body.empty;
     }
-
-    // TODO(keyonghan): improve branch fetching logic, like using cache, https://github.com/flutter/flutter/issues/53108
-    for (String branch in await config.flutterBranches) {
-      final BuildStatus buildStatus = await buildStatusService.calculateCumulativeStatus(branch: branch);
-      final GitHub github = githubService.github;
-      final List<GithubBuildStatusUpdate> updates = <GithubBuildStatusUpdate>[];
-      log.debug('Computed build result of $buildStatus');
-      // Insert build status to bigquery.
-      await _insertBigquery(buildStatus, branch);
-      final List<PullRequest> pullRequests = await githubService.listPullRequests(slug, branch);
-      for (PullRequest pr in pullRequests) {
-        final GithubBuildStatusUpdate update = await datastore.queryLastStatusUpdate(slug, pr);
-        if (update.status != buildStatus.githubStatus) {
-          log.debug('Updating status of ${slug.fullName}#${pr.number} from ${update.status}');
-          final CreateStatus request = CreateStatus(buildStatus.githubStatus);
-          request.targetUrl = 'https://flutter-dashboard.appspot.com/#/build';
-          request.context = config.flutterBuild;
-          if (!buildStatus.succeeded) {
-            request.description = config.flutterBuildDescription;
-          }
-
-          try {
-            await github.repositories.createStatus(slug, pr.head.sha, request);
-            update.status = buildStatus.githubStatus;
-            update.updates += 1;
-            update.updateTimeMillis = DateTime.now().millisecondsSinceEpoch;
-            updates.add(update);
-          } catch (error) {
-            log.error('Failed to post status update to ${slug.fullName}#${pr.number}: $error');
-          }
+    luciBuildService.setLogger(log);
+
+    final String repo = request.uri.queryParameters[fullNameRepoParam] ?? 'flutter/flutter';
+    final RepositorySlug slug = RepositorySlug.full(repo);
+
+    for (String branch in await config.getSupportedBranches(slug)) {
+      final Commit tipOfTreeCommit = Commit(sha: branch, repository: slug.fullName);
+      final SchedulerConfig schedulerConfig = await scheduler.getSchedulerConfig(tipOfTreeCommit);
+      final List<LuciBuilder> postsubmitBuilders =
+          await scheduler.getPostSubmitBuilders(tipOfTreeCommit, schedulerConfig);
+      final LuciService luciService = luciServiceProvider(this);
+      final Map<LuciBuilder, List<LuciTask>> luciTasks = await luciService.getRecentTasks(builders: postsubmitBuilders);
+
+      String status = GithubBuildStatusUpdate.statusSuccess;
```

Comment:
Should this be `GithubBuildStatusUpdate.statusFailure`?

If we have a 100 pending commits open, we should close the tree.

Reply:
It is already limited to 40 states per builder and handled in _getLatestStatus